lib/cluster.js

```diff
@@ -71,7 +90,18 @@ class AggregatorRegistry extends Registry {
 				type: GET_METRICS_REQ,
 				requestId
 			};
-			for (const id in cluster.workers) cluster.workers[id].send(message);
+      let failedWorkers = new Set();
+      for (const id of that.aliveWorkers) {
+        let worker = cluster.workers[id];
+        if (worker === undefined || worker.isDead() || !worker.isConnected()) {
+          failedWorkers.add(id);
+          request.pending --;
+          request.workerCount --;
+        } else {
+          worker.send(message)
+        }
+      }
+      that.aliveWorkers = new Set([...that.aliveWorkers].filter(x => !failedWorkers.has(x)));
```

Comment:
Can you just do `this.aliveWorkers.remove(...)` in the loop, instead of making a new `Set` instance each time?

Reply:
I'll have to check the semantics of mutating the Set while iterating, but yes will change.